src/eventMapping.ts

```diff
@@ -104,43 +104,45 @@ export function getCloudEvent(req: express.Request): CloudEventsContext | null {
     return cloudevent;
   }
 
-  console.log('Converting from background event to CloudEvent');
   const event = getBackgroundEvent(req);
-  if (event === null) {
-    console.error('Unable to extract background event');
-    return null;
+  if (!event) {
+    throw new Error('Unable to extract background event');
   }
 
   const context = event.context;
   const data = event.data;
 
+  cloudevent = {
+    contenttype: CE_CONTENT_TYPE,
+    id: context.eventId,
+    specversion: CE_SPEC_VERSION,
+    time: context.timestamp,
+    data: data,
+  };
+
   // Determine CloudEvent type attribute.
-  if (context.eventType === undefined) {
-    console.error('Unable to find background event type');
-    return null;
+  if (!context.eventType) {
+    throw new Error('Unable to find background event type');
   }
-  const ceType = typeBackgroundToCloudEvent[context.eventType];
+  const ceType = BACKGROUND_TO_CLOUDEVENT_EVENT_TYPE_MAP[context.eventType];
 
   // Determine CloudEvent service attribute.
-  if (context.resource === undefined) {
-    console.error('Unable to find background event resource');
-    return null;
+  if (!context.resource) {
```

Comment:
All of the property validation checks could be next to each other where possible. Comments shouldn't be needed given we have an error message.

```ts
if (!context.resource) throw new Error('Unable to find background event resource');
if (!context.eventType) throw new Error('Unable to find background event type');
...
```